src/hotspot/share/classfile/defaultMethods.cpp

```diff
@@ -1073,7 +1056,11 @@ static void merge_in_new_methods(InstanceKlass* klass,
   Array<int>* original_ordering = klass->method_ordering();
   Array<int>* merged_ordering = Universe::the_empty_int_array();
 
-  int new_size = klass->methods()->length() + new_methods->length();
+  int new_methods_length = klass->methods()->length() + new_methods->length();
+  if (new_methods_length > std::numeric_limits<u2>::max()) {
+    THROW_MSG(vmSymbols::java_lang_InternalError(), "default methods error methods overflow");
```

Comment:
That message doesn't quite read right to me - not clear exactly what has overflowed here.